datadotworld/slack-app
# [PAPPY 29] : Migrate and fix data.world Slack app integration

Addresses : https://dataworld.atlassian.net/browse/PAPPY-29

Testing : To test it locally and easily,
First run the server locally  
1) Install Postgres and start it. Create database called 'slackapp' on it. 
2) Install ngrok and bind ngrok to port 5000 (run ngrok http 5000). Keep ngrok url handy
3) Download this branch and create a .env file in the project directory(i.e slack-app folder).
4) Copy all env variables from here to .env file :  https://docs.google.com/document/d/1k_2ugXt9m-u0Lwl2DvMjI9Fosu9bvnj6sImtJqAq9oQ/edit
5) Replace <pg_password> with your postgres password. If you have different postgres username, change it as well. 
6) Replace <ngrok_url> with ngrok url of step 2. 
7) Start the server by running 'yarn dev' from project directory

Install and run ddw slack staging app in workspace 
- Create new workspace in slack to test slack app.  
-  Go to slack app manage distribution page(https://app.slack.com/app-settings/T09RTLM9B/A042EJVBFB6/distribute). 
   Note that you need to be app collaborator to access this page. 
- Click on 'Add to slack' button. 
- Select workspace created for testing in top right corner dropdown  and follow the prompt

To see what slack-app does, look [here](https://datadotworld.github.io/slack-app/)
